net-p2p/retroshare/retroshare-0.6.7.ebuild

```diff
@@ -0,0 +1,133 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit flag-o-matic desktop qmake-utils xdg
+
+DESCRIPTION="F2F, private and secure file sharing application"
+HOMEPAGE="https://retroshare.cc"
+SRC_URI="http://download.opensuse.org/repositories/network:/retroshare/Debian_Testing/retroshare-common_${PV}.orig.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/RetroShare"
+# REVIEW(setan): we could build from the link belowe to be closer to upstream
+# but this requires git modules
+# https://github.com/RetroShare/RetroShare/archive/refs/tags/{PV}.tar.gz
+
+LICENSE="AGPL-3 Apache-2.0 CC-BY-SA-4.0 GPL-2 GPL-3 LGPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="keyring cli +gui +jsonapi libupnp +miniupnp +service +sqlcipher plugins"
+
+REQUIRED_USE="
+	|| ( gui service )
+	?? ( libupnp miniupnp )
+	service? ( || ( cli jsonapi ) )"
+
+RDEPEND="
+	app-arch/bzip2
+	dev-libs/openssl:0=
+	>=dev-libs/rapidjson-1.1.0
+	sys-libs/zlib
+	keyring? ( app-crypt/libsecret )
+	gui? (
+		dev-qt/qtcore:5
+		dev-qt/qtmultimedia:5
+		dev-qt/qtnetwork:5
+		dev-qt/qtprintsupport:5
+		dev-qt/qtscript:5
+		dev-qt/qtxml:5
+		dev-qt/qtgui:5
+		dev-qt/qtwidgets:5
+		dev-qt/qtx11extras:5
+		x11-libs/libX11
+		x11-libs/libXScrnSaver
+	)
+	libupnp? ( net-libs/libupnp:= )
+	miniupnp? ( net-libs/miniupnpc:= )
+	service? ( dev-qt/qtcore:5 )
```

Comment:
With 0.6.7 qtcore is not needed anymore for running retroshare-service, while if qmake is used as build system (as is the case looking at the ebuild) it is still needed at compile time so please move this line to BDEPEND

Reply:
So from what I gather qtcore is needed as a build dependency for all, but for our use case it will be required at runtime only if we use the GUI?

Reply:
exactly

Reply:
upnps can stay here, right?
If so I will resolve this suggestion and focus on the one below.

Reply:
yeah UPNP libraries should stay there :)